APIs/JSON/controllers/http/base.js

```diff
@@ -0,0 +1,37 @@
+import BaseController from "@sama/common/controller.js"
+
+export default class BaseHttpController extends BaseController {
+  async parseJsonBody(res) {
+    return new Promise((resolve, reject) => {
+      let buffer = Buffer.alloc(0)
+      res.onData((chunk, isLast) => {
+        buffer = Buffer.concat([buffer, Buffer.from(chunk)])
+        if (isLast) {
+          try {
+            resolve(JSON.parse(buffer.toString()))
+          } catch (error) {
+            reject(new Error("Invalid JSON input"))
+          }
+        }
+      })
+
+      res.onAborted(() => reject(new Error("Request aborted")))
+    })
+  }
+
+  getCookie(req) {
+    return req.getHeader("cookie")
+  }
+
+  setRefreshTokenCookie(res, token) {
```

Comment:
can move this method to `HttpAuthController ` because it's called in HttpAuthController only